code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCardRulesEngine.java

```diff
@@ -72,6 +72,9 @@ Map<Surface, List<PropositionItem>> evaluate(@NonNull final Event event) {
                 continue;
             }
 
+            // store valid content card schema data in the ContentCardMapper for later use
+            ContentCardMapper.getInstance().storeContentCardSchemaData(propositionAsContentCard);
```

Comment:
Wondering if we should add this logic in `addOrReplaceContentCards` where the qualification happens. We can maintain a smaller mapping for qualified cards only instead of all cards returned by server. Thoughts?

Reply:
i think this is a good idea. i can make this change.